transport/transport_test.go

```diff
@@ -1640,13 +1640,6 @@ func TestInvalidHeaderField(t *testing.T) {
 	if err != nil {
 		return
 	}
-	opts := Options{
-		Last:  true,
-		Delay: false,
-	}
-	if err := ct.Write(s, nil, expectedRequest, &opts); err != nil && err != io.EOF {
```

Comment:
Instead of deleting the check, what if we replaced `io.EOF` with `errStreamDone` ?

Reply:
But the purpose of this check is to verify whether the write of the request succeed or not. `errorStreamDone` would also mean the write fails. And I am not seeing why it is necessary for the test, and that's how I decided to just delete it.

Reply:
I think you're right, the check in the first place wasn't serving any purpose. It was fine in both cases; success `err == nil` or failure `err == io.EOF`. The only purpose that it was serving then, was to make sure that write doesn't return any other error except io.EOF. Deleting this should fine then.